dropwizard-jersey/src/main/java/io/dropwizard/jersey/validation/FuzzyEnumParamConverterProvider.java

```diff
@@ -47,7 +56,35 @@ public T fromString(String value) {
                     return null;
                 }
 
-                final Enum<?> constant = Enums.fromStringFuzzy(value, constants);
+                if (fromStringMethod != null) {
+                    try {
+                        Object constant = fromStringMethod.invoke(null, value);
+                        // return if a value is found
+                        if (constant != null) {
+                            return (T) constant;
+                        }
+                        final String errMsg =
+                            String.format("%s is not a valid %s", parameterName, rawType.getSimpleName());
+                        throw new WebApplicationException(getErrorResponse(errMsg));
+                    } catch (IllegalAccessException e) {
+                        final String errMsg =
+                            String.format("Not permitted to call fromString on %s", rawType.getSimpleName());
+                        LOGGER.debug(errMsg, e);
+                        throw new WebApplicationException(getErrorResponse(errMsg));
+                    } catch (InvocationTargetException e) {
+                        if (e.getCause() instanceof WebApplicationException) {
+                            throw (WebApplicationException) e.getCause();
```

Comment:
I am only confused as to in what cases does this occur?

Reply:
In the APIs for our applications, we have a standardized format for error messages, including things like extended error codes, which are part of the API documentation. In the `fromString` implementation for an enum that is part of the API (it's only ever referenced by the resource class), we then throw a `WebApplicationException` with the correct extended error code.
The lines here ensures that that exception itself is propagated, and not wrapped, so that the error code etc. that the `fromString` implementation indicated are used in the response.